### PR TITLE
Don't remove wallet account if `storageAddress` is undefined

### LINKED ISSUE #541: Wallet connection breaks when switching from Pendulum to any other network
### Describe the bug
If a user switches the network from Pendulum to any of the other two networks and the wallet is connected then its diconnects and the user have to connect again, this happens only from Pendulum not for other networks

### Reproduction

1. Go to https://portal.pendulumchain.org/pendulum
2. Try switching network
3. See that the wallet connection breaks and the user have to connect again

### Expected behavior
The wallet should stay connected even when networks are switched

### Desktop (please complete the following information):

- OS: MacOS
- Browser: Chrome

### PR DESCRIPTION
**Pull request by ebma**
I think this was caused by some kind of race condition that sometimes happens when switching the new tenant. I think it's fine to remove that call to fix the problem.

Closes #541.

### CODE REVIEW

**Comment by ebma** [comment]
@prayagd when testing the deploy preview, please also consider checking how the wallet connection behaves for wallet connect etc.

**Review by prayagd — COMMENTED** [review]
Tested two different scenarios
- If i connect only to Pendulum network then the connection breaks when switching to other chains, is this expected?
- If i connect to all chains independently then the connection does not break
- Wallet connect does not work, screenshot of console attached
Screenshot 2024-10-22 at 10 52 45 AM

**Comment by ebma** [comment]
> If i connect only to Pendulum network then the connection breaks when switching to other chains, is this expected?

Hmm well I can reproduce that it breaks on the deploy preview. It worked for me locally though. I'll have a look again.

> Wallet connect does not work, screenshot of console attached

Right, sorry. We need to whitelist all the domains on our walletconnect account. I whitelisted the domain of this deploy preview now so it should work. I'll let you know when I looked into the other bug again so you can test.

File: src/GlobalStateProvider/index.tsx [hunk 1]
```diff
@@ -48,15 +48,16 @@ const GlobalStateProvider = ({ children }: { children: JSX.Element }) => {
     set,
     clear,
   } = useLocalStorage<string | undefined>({
-    key: `${storageKeys.ACCOUNT}-${tenantName}`,
-    expire: EXPIRATION_PERIOD,
+    key: `${storageKeys.ACCOUNT}`,
+    expire: undefined,
```
[the remark this answers is not in the document]
Comment (ebma) [reply src/GlobalStateProvider/index.tsx:52]: I set it to `expire: undefined` to highlight that we could set an expiry but deliberately choose not to. I can also remove it though, you are right.

File: src/GlobalStateProvider/index.tsx [hunk 2]
```diff
@@ -71,21 +72,22 @@ const GlobalStateProvider = ({ children }: { children: JSX.Element }) => {
...
         return;
       }
       // skip if tenant already initialized
-      if (tenantRef.current === tenantName || accountAddress) return;
+      if (tenantRef.current === tenantName) return;
       tenantRef.current = tenantName;
-      const selectedWallet = await initSelectedWallet(dAppName, tenantName, storageAddress);
-      if (selectedWallet) setWallet(selectedWallet);
+      // Delay this to allow the wallet extension to be injected, otherwise the call to wallet.enable() might fail
+      setTimeout(async () => {
```
[the remark this answers is not in the document]
Comment (ebma) [reply src/GlobalStateProvider/index.tsx:84]: Agreed, I'll change it 🫡

File: src/GlobalStateProvider/index.tsx [hunk 3]
```diff
@@ -48,15 +48,16 @@ const GlobalStateProvider = ({ children }: { children: JSX.Element }) => {
...
+    key: `${storageKeys.ACCOUNT}`,
+    expire: undefined,
   });
 
   const clearLocalStorageWallets = () => {
     storageService.remove(LocalStorageKeys.SELECTED_WALLET_NAME);
   };
 
   const removeWalletAccount = useCallback(async () => {
+    console.log('Removing wallet account', walletAccount);
```
[the remark this answers is not in the document]
Comment (ebma) [reply src/GlobalStateProvider/index.tsx:60]: Probably not, I'll remove it.

**Review by Sharqiewicz — APPROVED** [review]
Well done! ✅ It works. I hope Talisman will address this in the future, so we won’t need to rely on setTimeout, because using setTimeout is not ideal

**Comment by ebma** [comment]
@prayagd can you also try again and approve if it works?

**Review by prayagd — APPROVED** [review]
LGTM, we can merge

**Pull request merged**